### ISSUE TITLE
not wrapped multiple assertions

### ISSUE THREAD

**Issue by l3r8yJ** [volodya-lombrozo/jtcop#499]
**I suggest adding a new rule to enforce wrapping assertions in `assertAll`.** Below are examples to illustrate this:

**Bad:**  
```java
@Test
void doesSomethingCorrectly() {
  assertEquals(actual, expected);
  assertThat(actual, hasSize(expected.size));
}
```

**Good:**  
```java
@Test
void doesSomethingCorrectly() {
  assertAll(
    () -> assertEquals(actual, expected),
    () -> assertThat(actual, hasSize(expected.size))
  );
}
```

**Why is this better?**  
Wrapping assertions in `assertAll` ensures all failures are reported **immediately** in a single test run. Without it, the test stops at the first failed assertion, forcing you to fix errors one at a time. This promotes better debugging efficiency and test reliability.

@volodya-lombrozo wdyt?

**Comment by volodya-lombrozo** [comment]
@l3r8yJ It seems that the main issue is the number of `assert` statements. Ideally, a single unit test should have exactly one `assert` [statement](https://softwareengineering.stackexchange.com/questions/7823/is-it-ok-to-have-multiple-asserts-in-a-single-unit-test):

> I don't think it's necessarily a bad thing, but I do think we should strive towards only having single asserts in our tests. This means you write a lot more tests and our tests would end up testing only one thing at a time.

 We need to have exactly one `assert` statement, you are right, but I don't think it's a good idea to force developers to use exactly `assertAll` function for that. 

Let's create a check that would require a single `assert` statement for a single unit test? That way, developers can decide for themselves how to resolve it. What do you think?

**Comment by l3r8yJ** [comment]
@volodya-lombrozo sounds reasonable, sure, let's do rule for single assertion
